### ISSUE TITLE
Query loop filters removed on saving theme

### ISSUE THREAD

**Issue by FORTE-WP** [WordPress/create-block-theme#734]
I have made some custom templates for single posts within a specific category. The templates also have a query loop with a filter for that same category, so visitors can easily go to the next similar post. But when I save the theme, the category filters are removed. The templates show all posts.

Steps:

1. Create a custom template
2. Include the query loop block
3. In Settings, create a new search and have Post as post type.
4. Go to the Filter section and enter a specific category taxonomy
5. Go to the Create Block Theme section in the Site Editor. Select Save Changes to Theme.
6. Select all items, besides ‘Localize text’ and ‘Remove Navigation Refs’ and save the changes
7. Go to the custom template to see that the filter is removed.

WordPress: 6.6.2, Create Block Theme: 2.5.0

**Comment by sahil-gidwani** [comment]
Hey @FORTE-WP , I faced a similar issue and upon further review it seems like this was intentional.
Please refer to the [code](https://github.com/WordPress/create-block-theme/blob/33c5ff9ad4b762a159848a22228269e117c82dbd/includes/create-theme/theme-templates.php#L323) here in `theme-templates.php`.

**Comment by FORTE-WP** [comment]
Indeed, it seems intentional. However, I believe it should be an option, similar to the nav reference. When exporting to a ZIP file, the taxonomy filter can be removed, but saving the theme should not break it.

**Comment by pbking** [comment]
Indeed, I believe it was intentional but shouldn't be.  I don't think query IDs should be removed at all (optional or not). They should be unique per-page but otherwise (unlike other properties removed when parsing) should remain in templates/patterns.

**Comment by jtleathers** [comment]
The option to keep taxquery would be so nice. I forget that this plugin removes them and always have to go back and manually add them when checking the front-end and seeing that it didn't work.

**Comment by colrmaptime** [comment]
After beating my head against my laptop I finally came to Git Hub only to see that this is intentional? Darn, not sure how to handle this moving forward. I'm using taxonomy filtering in my query loops and it is stripping them out on saving. Can I hard code these in somewhere?

**Comment by pbking** [comment]
The problem with that is that the ID of the taxonomy you are using is not guaranteed (likely?) to be the same from site to site.  So if you design a theme and use taxonomy filtering the taxonomy ID in the template might be `44` but in the production site the taxonomy you're trying to use is `323`.  Since taxonomy filtering is done with ids instead of slugs it's not really possible to make that portable.

The solution is for taxonomy filtering to be slug-based instead of id-based and that would require changes to core or possibly a little plugin to add that ability to the query block.

**Comment by jtleathers** [comment]
@pbking I think the misunderstanding here comes down to the assumption that themes are always built to be used on multiple sites. In many cases they are not and are created for a specific site but we want to be able to save the files for Git version control instead of WordPress database version control. In these cases this plugin actually causes problems and it's not just that it is missing a feature. 

For instance, I can build the template how I want with the taxonomy filters and copy the code from the editor and paste it into the template file directly. The problem is, when I choose "Process Only Modified Templates" it will _still_ override the code I copied and pasted. So every time I "Save changes to theme", this plugin undoes the work I have already done. At that point, I'm better off not using this plugin and instead copying/pasting the code from the editor into template files.

If instead a checkbox was added to _not_ strip filter settings, I could use this plugin again without it undoing my work.

**Comment by colrmaptime** [comment]
Thank you for clarifying even further @jtleathers what my issue is exactly. I appreciate you mentioning that some are actually using this plugin perhaps for only one site's use.

**Comment by pbking** [comment]
Yup, I understand what you're saying and it's a valid point.  I agree that it should be optional for CBT to remove that.  It's causing pain for you.  I also use CBT nearly exclusively for "one shot" themes.  It was originally a tool built for Designers at Automattic to build themes to distribute but that's not how I use it anymore either.

But the way I work is to do that work locally and then send that theme to a production (or staging) site where the IDs of things are very often not the same, so making that same assumption caused me some pain.  Which was the only reason I mentioned it.

If I ever find time to work on Create Block Theme again I'll be happy to implement an optional checkbox for that.

**Comment by jtleathers** [comment]
@pbking Actually, there is already [a Pull Request for this feature](https://github.com/WordPress/create-block-theme/pull/743) from last year.

**Comment by pbking** [comment]
Thanks for the heads up.  I just sat down to do it and that was a nice little find.
